auth/serviceaccount.go

```diff
@@ -0,0 +1,60 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/kelseyhightower/envconfig"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type ServiceAccount struct {
+	UserId string
+}
+
+type ServiceAccountAuthenticator struct {
+	TokenEndpoint string `envconfig:"TIDEPOOL_AUTH_SERVICE_TOKEN_ENDPOINT"`
+}
+
+func NewServiceAccountAuthenticator() (*ServiceAccountAuthenticator, error) {
+	s := &ServiceAccountAuthenticator{}
+	err := envconfig.Process("", s)
+	return s, err
+}
+
+func (s *ServiceAccountAuthenticator) GetServiceAccount(ctx context.Context, clientId, clientSecret string) (*ServiceAccount, error) {
```

Comment:
This function has no test coverage.

Given what it does, I don't think a test is necessary. Maybe the next time you're changing this code, consider throwing this a test that checks that the returned ServiceAccount's UserId isn't blank (that's really about the only logic in here).

The test coverage of authentication.go is also 0, and that seems bad, but out of scope for this PR.